src/psycopmlutils/loaders/non_numerical_coercer.py

```diff
@@ -29,15 +29,21 @@ def multiply_inequalities_in_df(
     in_eqs = sorted(ineq2mult.keys(), key=len, reverse=True)
 
     for in_eq in in_eqs:
-        starts_with_ineq_idxs = df[col_to_multiply].str.startswith(in_eq).fillna(False)
+        try:
+            starts_with_ineq_idxs = (
+                df[col_to_multiply].str.startswith(in_eq).fillna(False)
+            )
+        except AttributeError:
+            # If the column is no longer a string (i.e. all values have been coerced), continue
+            continue
 
         df.loc[starts_with_ineq_idxs, col_to_multiply] = (
             df.loc[starts_with_ineq_idxs, col_to_multiply]
-            .str.replace(in_eq, "")
-            .str.replace(",", ".")  # Replace comma with dot for float conversion.
+            .str.replace(",", ".")
+            .str.extract(r"(\d+\.\d+|\d+)", expand=False)
             .astype(float)
             .mul(ineq2mult[in_eq])
-            .round(0)
+            .round(2)
```

Comment:
Any specific reason why we round the value?

Reply:
This doesn't seem like an inspection function. I would remove it as it is essentially a loss of information. It probably doesn't matter, though. (interesting test, though - does rounding all numbers where we believe decimals shouldn't be relevant cause a decrease in performance).